libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -36,21 +36,25 @@ import Test.QuickCheck (shrinkList)
 
 instance Ord a => Sized (Map.Map a b) where
   sizeOf = toInteger . Map.size
-  liftSizeSpec sz = TypeSpec (MapSpec mempty mempty (typeSpec sz) TrueSpec NoFold) []
-  liftMemberSpec xs = typeSpec (MapSpec mempty mempty (MemberSpec xs) TrueSpec NoFold)
-  sizeOfTypeSpec (MapSpec mustk mustv size _ _) =
+  liftSizeSpec sz = typeSpec $ defaultMapSpec {mapSpecSize = typeSpec sz}
+  liftMemberSpec xs = typeSpec $ defaultMapSpec {mapSpecSize = MemberSpec xs}
+  sizeOfTypeSpec (MapSpec _ mustk mustv size _ _) =
     typeSpec (atLeastSize (sizeOf mustk))
       <> typeSpec (atLeastSize (sizeOf mustv))
       <> size
 
 data MapSpec fn k v = MapSpec
-  { mapSpecMustKeys :: Set k
+  { mapSpecHint :: Maybe Integer
+  , mapSpecMustKeys :: Set k
   , mapSpecMustValues :: [v]
   , mapSpecSize :: Specification fn Integer
   , mapSpecElem :: Specification fn (k, v)
   , mapSpecFold :: FoldSpec fn v
   }
 
+defaultMapSpec :: Ord k => MapSpec fn k v
+defaultMapSpec = MapSpec Nothing mempty mempty TrueSpec TrueSpec NoFold
+
```

Comment:
Might be worth a comment, that this is 'emptySpec', but without the HasSpec constraint.